server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -334,15 +334,34 @@ public boolean isInSortOrderExecutionRequired() {
         }
 
         /**
-         * Return false if this aggregation or any of the child aggregations does not support concurrent search
+         * Return false if this aggregation or any of the child aggregations does not support concurrent search.
+         * As a result, such aggregation will always be executed sequentially despite concurrency is enabled for the query phase.
+         * Note: aggregations that don't support concurrency, may or may not support offloading their collection to the search worker threads,
+         * depending on what {@link #supportsOffloadingSequentialCollection()} returns.
          */
-        public boolean supportsConcurrentExecution() {
+        // TODO this is not used yet, it will be used once we enable parallel collection for the query phase
+        public boolean supportsParallelCollection() {
```

Comment:
I renamed this to hopefully be more accurate about what it controls, especially as I introduced the new supportsOffloadingSequentialCollection below.